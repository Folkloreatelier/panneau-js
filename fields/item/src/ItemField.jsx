import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { FormGroup } from '@panneau/field';

const propTypes = {
    name: PropTypes.string,
    label: PropTypes.string,
    value: PropTypes.string,
    onChange: PropTypes.func,
};

const defaultProps = {
    name: null,
    label: null,
    value: null,
    onChange: null,
};

class ItemField extends Component {
    static parse(value) {
        return value;
    }

    constructor(props) {
        super(props);

        this.onChange = this.onChange.bind(this);

        this.refInput = null;
    }

    onChange(e) {
        const newValue = (e.currentTarget || this.refInput).value;
        if (this.props.onChange) {
            this.props.onChange(newValue);
        }
    }

    render() {
        const {
            label,
            name,
            value,
            ...other
        } = this.props;

        return (
            <FormGroup
                className="form-group-item"
                name={name}
                label={label}
                {...other}
            >
                <input
                    type="text"
                    value={value}
                    onChange={this.onChange}
                    ref={(ref) => { this.refInput = ref; }}
                />
            </FormGroup>
        );
    }
}

ItemField.propTypes = propTypes;
ItemField.defaultProps = defaultProps;

export default ItemField;
