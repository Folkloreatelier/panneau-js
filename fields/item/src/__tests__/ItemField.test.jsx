/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import renderer from 'react-test-renderer';
import { IntlProvider } from 'react-intl';
import ItemField from '../ItemField';

test('match snapshot', () => {
    const component = renderer.create((
        <IntlProvider locale="en">
            <ItemField
                label="Label"
                value={{}}
            />
        </IntlProvider>
    ));
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
