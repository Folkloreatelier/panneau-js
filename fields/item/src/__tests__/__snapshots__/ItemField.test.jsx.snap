// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<div
  className="form-group undefined"
>
  <div
    className="form-group-label"
  >
    <label
      className="control-label"
      htmlFor={null}
    >
      Label
    </label>
  </div>
  <div
    className="form-group-inner"
    style={
      Object {
        "display": "block",
      }
    }
  >
    <div
      className="mb-2 undefined"
    >
      <div
        className="card"
      >
        <div
          className="card-body"
        >
          <div
            className="undefined"
          >
            <div
              className="undefined"
            >
              <div
                className="undefined"
              >
                <div
                  className="undefined"
                >
                  <div
                    className={undefined}
                  />
                </div>
                <div
                  className="undefined"
                >
                  <div
                    className="undefined"
                  >
                    <div
                      className="btn-group"
                    >
                      <button
                        className="btn btn-light fas fa-times"
                        onClick={[Function]}
                        type="button"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
